### PR TITLE
Embed cron, dmon, and trigger iden into vars (SYN-3224)

### PR DESCRIPTION
**Pull request by vEpiphyte**
Embed idens at runtime as `$_iden` for cron, dmon and trigger systems.

### CODE REVIEW

File: synapse/lib/agenda.py [hunk 1]
```diff
@@ -856,7 +856,7 @@ async def _runJob(self, user, appt):
             starttime = self._getNowTick()
             success = False
             try:
-                opts = {'user': user.iden, 'view': appt.view}
+                opts = {'user': user.iden, 'view': appt.view, 'vars': {'_iden': appt.iden}}
```
Comment (mikemoritz) [inline synapse/lib/agenda.py:859]: Is there a chance in the future we would want to stuff more info in, and then maybe it makes sense to take the hit on one  var (like a $_self) that's a dict?

Comment (invisig0th) [inline synapse/lib/agenda.py:859]: If we dont end up with something like `$auto.iden` these should be converted `_iden` -> `croniden` ( et al )

Comment (invisig0th) [reply synapse/lib/agenda.py:859]: Maybe? We already have the `$tag` and `$prop` being mapped in on triggers though, so we'd end up with either duplicates or a schism, but i could def see `$auto.iden` maybe being useful as the "unified way to represent the iden of an automation element"

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/trigger.py [hunk 2]
```diff
@@ -359,8 +359,12 @@ async def _execute(self, node, vars=None, view=None):
             'user': self.user.iden,
         }
 
-        if vars is not None:
-            opts['vars'] = vars
+        if vars is None:
+            vars = {}
+        else:
+            vars = vars.copy()
+        vars['auto'] = {'iden': self.iden, 'type': 'trigger'}
```
Comment (invisig0th) [inline synapse/lib/trigger.py:366]: Does this need to be a setdefault() just in case?

Comment (vEpiphyte) [reply synapse/lib/trigger.py:366]: Triggers do not do capture of user variables. Their vars are populated prior to execution; but I do need to move this vars population up one level upon review.

File: synapse/tests/test_cortex.py [hunk 3]
```diff
@@ -852,19 +853,29 @@ async def test_cortex_storm_dmon_log(self):
...
-            iden = await core.callStorm('''
-                $que = $lib.queue.add(foo)
-
-                $ddef = $lib.dmon.add(${
-                    $lib.print(hi)
-                    $lib.warn(omg)
-                    $que = $lib.queue.get(foo)
-                    $que.put(done)
-                })
+            with self.getStructuredAsyncLoggerStream('synapse.storm.log',
```
Comment (mikemoritz) [inline synapse/tests/test_cortex.py:856]: did you want to doc $auto in the automation section (or somewhere)?

Comment (vEpiphyte) [reply synapse/tests/test_cortex.py:856]: I'll add some docs.

Comment (vEpiphyte) [reply synapse/tests/test_cortex.py:856]: added

File: docs/synapse/userguides/storm_ref_automation.rstorm [hunk 4]
```diff
@@ -135,6 +135,17 @@ The "one-time" nature of triggers may result in a small number of edge cases whe
...
+**Variables:**
+
+Triggers automatically have the Storm variable ``$auto`` populated when they run. The ``$auto`` variable is a dictionary which contains the following keys:
+
+  ``$auto.iden``
+    The identifier of the Trigger.
+
+  ``$auto.type``
+    The type of automation. For a Trigger this value will be ``dmon``.
+
```
Comment (invisig0th) [inline docs/synapse/userguides/storm_ref_automation.rstorm:147]: Do we wanna take this opportunity to plumb and document something like `$auto.opts.tag` (etc) for the other vars ( and leave the existing ones in place for now obvs ) ?

Comment (vEpiphyte) [reply docs/synapse/userguides/storm_ref_automation.rstorm:147]: We certainly could. Each trigger through typoe would end up with its own opts dict based on what was done. This is also a good opportunity to at least through in a `$form` and `$valu` key so that is universally available across triggers.

**Review by invisig0th — APPROVED** [review]

**Review by mikemoritz — APPROVED** [review]
